test/use-swr.test.tsx

```diff
@@ -543,41 +543,41 @@ describe('useSWR - refresh', () => {
       return new Promise(res => setTimeout(res, 150))
     })
     expect(container.firstChild.textContent).toMatchInlineSnapshot(`"count: 2"`)
+    fireEvent.click(container.firstElementChild)
     await act(() => {
-      fireEvent.click(container.firstElementChild)
       // it will clear 200ms timer and setup a new 300ms timer
       return new Promise(res => setTimeout(res, 200))
     })
-    expect(container.firstChild.textContent).toMatchInlineSnapshot(`"count: 2"`)
+    expect(container.firstChild.textContent).toMatchInlineSnapshot(`"count: 3"`)
     await act(() => {
       return new Promise(res => setTimeout(res, 110))
     })
-    expect(container.firstChild.textContent).toMatchInlineSnapshot(`"count: 3"`)
```

Comment:
@promer94 would you mind taking a look on those changes on this test case to see if there's any concern or mistakes? found they were created by you in #207 🙏

Reply:
The behavior is changed here. When the refreshInterval changes after revalidating, the previous timer will not be clear because
```js
if (timer) {
  timer = null
  // clearTime wont take any effect here
  clearTimeout(timer)
}
```

I think it would be batter if we could keep the same behavior by

```js
if (timer) {
  clearTimeout(timer)
  timer = null
}
```

Reply:
oooh thanks for catching my obvious mistake 🙈👍